src/main/java/com/redhat/composer/services/RetrieveService.java

```diff
@@ -19,9 +30,16 @@
 @ApplicationScoped
 public class RetrieveService {
 
+  // Should these be configurable by assistant? Should it change to tokens?
+  private static final int MAX_SEGMENT_SIZE_IN_CHARS = 4000;
```

Comment:
Not sure about the tokens thing. But it should be configurable.

I think optimally we have a configurable TOKENS count and a configurable CHARS count and we just prioritize over the other and document it. |

I.E. if both TOKENS and CHARS are set we ignore the CHARS value (or vice a versa)

Reply:
I made this configurable at the application level as well, also with space to override dynamically. I don't know how to create a proper Tokenizer for this, so even though there is logic to choose between chars vs tokens, it'll always just do characters until we add that. That can be something we enhance later, possibly with a selection of available tokenizers.